src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,53 +1,134 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
+interface Props {
+  onAdd?: (newMovie: Movie) => void,
+}
+
+export const NewMovie: React.FC<Props> = ({
+  onAdd = () => {},
+}) => {
+  const [newFilm, setnewFilm] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
   const [count] = useState(0);
 
+  const clearFields = () => {
+    setnewFilm({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const areFieldsFilled = () => {
+    return (newFilm.title) && (newFilm.imgUrl)
+    && (newFilm.imdbUrl) && (newFilm.imdbId);
```

Comment:
```suggestion
    return newFilm.title && newFilm.imgUrl
    && newFilm.imdbUrl && newFilm.imdbId;
```

brackets are redundant here
also, trim all values here, because now I can fill each field with spaces only